### PR TITLE
ci: add react ssr

### PR DESCRIPTION
**Pull request by daine**
# Summary | Résumé

PR to add the react ssr package

## Canary release for the React SSR Package
This is an implementation of the [Canary Release Plan](https://docs.google.com/document/d/14TwlBuSd01zNkF7srl8cL1KWP0dlMvWEiDZlrbY0F6c/edit)

Checklist:
- [x] `gcds-components-react-ssr/package.json` has the correct canary version format `0.22.2-canary.0` as of July 24, 2024
- [ ] Dry run of `npm publish` has been tested
- [x] StencilJS version needs to be pinned to `4.11.0` since we have a patch dependency on it, which we will have to update once we upgrade.
- [x] Appropriate patch has been applied #570
- [x] Added fixes that may have been missed from #535
- [x] Add new workspace build #600 
- [x] Add workspace to release manifest
- [x] Update components to the latest version (0.22.4)

## Potential impacts:
| Impact  | Notes |
| ------------- | ------------- |
| Publish Packages runs multiple times  | Tested this and NPM will not let it happen -- it will error out <br> `🙅‍♀️ <packagename> already published.`|
| Release please not detecting canary version bump  | Will need to be on the lookout for this, it will be too hard to test multiple releases so we will have to accommodate and fix it when it happens |

### CODE REVIEW

**Comment by daine** [comment]
Tested the CI run here for publishing:
https://github.com/daine/gcds-components/actions/runs/10071411555/job/27841892005
Screenshot 2024-07-24 at 10 52 35 AM

I commented out everything else but the react-ssr package. It was successful, so I then tested that package on the NextJS PR and it works on dev and builds.

File: packages/react-ssr/package.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  "scripts": {
+    "prebuild": "del-cli --force ./dist && copyfiles -f ../web/dist/components/*.js ./dist/esm/components && copyfiles -f ../../node_modules/@stencil/core/internal/client/index.js ./dist/esm/lib/stencil && copyfiles -f ../../node_modules/@stencil/core/internal/client/shadow-css.js ./dist/esm/lib/stencil && make-dir ./dist/esm && make-dir ./dist/types",
+    "build": "npm run build:esm && npm run build:types && npm run build:cjs",
+    "build:esm": "tsx ./scripts/build.ts",
+    "build:types": "tsc -p .",
+    "build:cjs": "esbuild dist/esm/*.js dist/esm/**/*.js dist/esm/**/**/*.js --log-level=error --outdir=dist/cjs --format=cjs --loader:.js=jsx",
+    "lint": "eslint ."
+  },
+  "dependencies": {
+    "@cdssnc/gcds-components": "^0.22.0",
```
Comment (ethanWallace) [inline packages/react-ssr/package.json:54]: Can you up the web components package dependency version to the latest version? `0.22.2`

**Review by ethanWallace — CHANGES_REQUESTED** [review]
Just one small thing

**Review by ethanWallace — DISMISSED** [review]
LGTM

**Review by melaniebmn — APPROVED** [review]
LGTM - ship it! Wohoooo

**Review by ethanWallace — APPROVED** [review]
LGTM!

**Pull request merged**